metroae

```diff
@@ -691,20 +691,23 @@ function pull {
         if [ -w "$SCRIPT_LOG_FILE" ]; then
             debug "${FUNCNAME[0]}: Script log is writable"
             sudo docker pull $METRO_AE_IMAGE:$MAX_CONTAINER_VERSION | tee -a $SCRIPT_LOG_FILE
+            ((status=$?+$PIPESTATUS))
         else
             if [ -z $METROAE_WRITE_WARNING_PRINTED ]; then
                 print_write_permission_warning
                 METROAE_WRITE_WARNING_PRINTED=1
+                status=$?
             fi
             debug "${FUNCNAME[0]}: Script log is not writable"
             sudo docker pull $METRO_AE_IMAGE:$MAX_CONTAINER_VERSION
+            status=$?
         fi
     else
         debug "${FUNCNAME[0]}: Script log is not found"
         sudo docker pull $METRO_AE_IMAGE:$MAX_CONTAINER_VERSION
+        status=$?
     fi
 
-    status=$?
     if [[ $status -ne 0 ]]; then
```

Comment:
The issue we're seeing today is that if we hit the container pull limit, we are not going into this code block. That means the status code must be 0, otherwise we would exit. My interpretation of this is that docker pull is returning 0 even if we hit the pull limit instead of a nonzero return code. If this is true, then would the changes you made prevent this from happening? Are we seeing a return code of 0 because we we are getting the return code from something other than docker pull?

Reply:
Vandan ran tests of docker pull on the command line. He found that docker pull returned a non-zero return code when there was a failure. But, when he added ' | tee " to the command, as is done inside the metroae script, the shell returned 0 because the last command executed was 'tee'. This is one of the things we must be careful of when using pipe. The last command return code supercedes any return codes that precede it in the pipe string.

Reply:
<img width="589" alt="exit_code_issue" src="https://user-images.githubusercontent.com/90286472/134838152-9463a480-dce5-44f7-a55a-3605c6bf198f.PNG">